### PR TITLE
Adds preliminary routes to backend api

### PR DESCRIPTION
**Pull request by eharley19**
This PR adds preliminary routes to backend api.

### CODE REVIEW

File: web_backend/api.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+def setup_db():
+    init_db()
+
+
+@app.teardown_appcontext
+def shutdown_session(exception=None):
+    db_session.remove()
+
+
+@app.route('/')
```
Comment (datdo) [inline web_backend/api.py:19]: remove if not being used.

File: web_backend/api.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from flask import Flask, jsonify
```
Comment (datdo) [inline web_backend/api.py:1]: you should move these methods into separate blueprints for ease of maintenance.

File: web_backend/api.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    db_session.remove()
+
+
+@app.route('/')
+def index():
+    primary_docs = PrimaryDoc.query.all()
+    return jsonify(primary_docs)
+
+
+@app.route('/company/<company_id>')
```
Comment (datdo) [inline web_backend/api.py:25]: don't start the route with a backslash when you move to blueprint

File: web_backend/api.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+def get_company(company_id):
+    primary_doc = PrimaryDoc.query.filter(PrimaryDoc.company_name.like(f"{company_id}%"))
```
Comment (datdo) [inline web_backend/api.py:27]: company_id is not name. It is the CIK.

File: web_backend/api.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+
+@app.route('/')
+def index():
+    primary_docs = PrimaryDoc.query.all()
+    return jsonify(primary_docs)
+
+
+@app.route('/company/<company_id>')
+def get_company(company_id):
```
Comment (datdo) [inline web_backend/api.py:26]: what if the company_id (the cik) does not exist in the database?

File: web_backend/database.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+
+connection_string = os.environ.get('DB_CONNECTION_STRING')
+engine = create_engine(connection_string, convert_unicode=True)
+db_session = scoped_session(sessionmaker(autocommit=False, autoflush=False, bind=engine))
+
+Base = declarative_base()
+Base.query = db_session.query_property()
```
Comment (datdo) [inline web_backend/database.py:12]: what's the point of this line?

File: web_backend/database.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def init_db():
+    from models import PrimaryDoc, Infotable
+    Base.metadata.create_all(bind=engine)
```
Comment (datdo) [inline web_backend/database.py:17]: add newline at end of file.

**Review by datdo — CHANGES_REQUESTED** [review]

File: web_backend/database.py (hunk 6 above)
Comment (TheNorthEestern) [reply web_backend/database.py:11]: https://stackoverflow.com/a/42309468

File: web_backend/filingapi/models.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+
+class Infotable(db.Model):
+    """Define Infotable Table"""
+    __tablename__ = 'infotable'
+    row_id = db.Column(db.String, primary_key=True)
+    accession_no = db.Column(db.String, db.ForeignKey('primary_doc.accession_no'))
+    cik_no = db.Column(db.String)
+    nameOfIssuer = db.Column(db.String)
+    titleOfClass = db.Column(db.String)
```
Comment (TheNorthEestern) [inline web_backend/filingapi/models.py:36]:
```suggestion
    name_of_issuer = db.Column(db.String)
    title_of_class = db.Column(db.String)
```

File: web_backend/filingapi/models.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    cusip = db.Column(db.String)
+    value = db.Column(db.Numeric)
+    sshPrnamt = db.Column(db.Integer)
+    sshPrnamtType = db.Column(db.String)
+    putCall = db.Column(db.String)
+    investmentDiscretion = db.Column(db.String)
+    otherManager = db.Column(db.String)
+    votingAuthority_Sole = db.Column(db.Integer)
+    votingAuthority_Shared = db.Column(db.Integer)
+    votingAuthority_None = db.Column(db.Integer)
```
Comment (TheNorthEestern) [inline web_backend/filingapi/models.py:46]:
```suggestion
    ssh_prnamt = db.Column(db.Integer)
    ssh_prnamt_type = db.Column(db.String)
    put_call = db.Column(db.String)
    investment_discretion = db.Column(db.String)
    other_manager = db.Column(db.String)
    voting_authority_Sole = db.Column(db.Integer)
    voting_authority_Shared = db.Column(db.Integer)
    voting_authority_None = db.Column(db.Integer)
```

File: web_backend/filingapi/models.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        return result.hexdigest()
+
+
+@dataclass
+class PrimaryDoc(db.Model):
+    accession_no: str
+    cik_no: str
+    company_name: str
+    filing_date: db.Date
+    infotable_rows: Infotable
```
Comment (datdo) [inline web_backend/filingapi/models.py:74]: should this be a collection of Infotable objects?

File: web_backend/filingapi/models.py [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    sshPrnamtType = db.Column(db.String)
+    putCall = db.Column(db.String)
+    investmentDiscretion = db.Column(db.String)
+    otherManager = db.Column(db.String)
+    votingAuthority_Sole = db.Column(db.Integer)
+    votingAuthority_Shared = db.Column(db.Integer)
+    votingAuthority_None = db.Column(db.Integer)
+
+    def __repr__(self):
+        return "<User(accession_no='%s', cik_no='%s', nameOfIssuer='%s', " \
```
Comment (datdo) [inline web_backend/filingapi/models.py:34]: change user to Infotable

File: web_backend/filingapi/models.py [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    """Define PrimaryDoc Table"""
+    __tablename__ = 'primary_doc'
+    accession_no = db.Column(db.String, primary_key=True)
+    cik_no = db.Column(db.String)
+    company_name = db.Column(db.String)
+    filing_date = db.Column(db.Date)
+    infotable_rows = db.relationship("Infotable")
+
+    def __repr__(self):
+        return "<User(accession_no='%s', cik_no='%s', filing_date='%s', company_name='%s')>" % (
```
Comment (datdo) [inline web_backend/filingapi/models.py:85]: change User to PrimaryDoc

**Review by datdo — CHANGES_REQUESTED** [review]

**Review by datdo — APPROVED** [review]

**Pull request closed without merging**